core/state/state_object.go

```diff
@@ -197,24 +208,48 @@ func (s *StateObject) GetState(db Database, key common.Hash) common.Hash {
 // GetCommittedState retrieves a value from the committed account storage trie.
 func (s *StateObject) GetCommittedState(db Database, key common.Hash) common.Hash {
 	// If the fake storage is set, only lookup the state here(in the debugging mode)
+	start := time.Now()
+	hitInCache := false
+
+	defer func() {
+		routeid := cachemetrics.Goid()
+		isSyncMainProcess := cachemetrics.IsSyncMainRoutineID(routeid)
+		isMinerMainProcess := cachemetrics.IsMinerMainRoutineID(routeid)
+		if isSyncMainProcess && hitInCache {
+			cachemetrics.RecordCacheDepth("CACHE_L1_STORAGE")
+			cachemetrics.RecordCacheMetrics("CACHE_L1_STORAGE", start)
+			cachemetrics.RecordTotalCosts("CACHE_L1_STORAGE", start)
+		}
+
+		if isMinerMainProcess && hitInCache {
+			cachemetrics.RecordMinerCacheDepth("MINER_L1_STORAGE")
+			cachemetrics.RecordMinerCacheMetrics("MINER_L1_STORAGE", start)
+			cachemetrics.RecordMinerTotalCosts("MINER_L1_STORAGE", start)
+		}
```

Comment:
Can we record a total IO cost here? no matter hit or miss.

Reply:
And I think the `L1 cache` metrics should measure from:
```
// GetState retrieves a value from the account storage trie.
func (s *StateObject) GetState(db Database, key common.Hash) common.Hash {
	// If the fake storage is set, only lookup the state here(in the debugging mode)
	if s.fakeStorage != nil {
		return s.fakeStorage[key]
	}
	// If we have a dirty value for this state entry, return it
	value, dirty := s.dirtyStorage[key]
	if dirty {
		return value
	}
	// Otherwise return the entry's original value
	return s.GetCommittedState(db, key)
}
```

Reply:
I think `dirtyStorage` belongs to L1 cache too.